libhdf5/hdf5dim.c

```diff
@@ -166,6 +166,16 @@ NC4_inq_dim(int ncid, int dimid, char *name, size_t *lenp)
                of records from all the vars that share this
                dimension. */
             *lenp = 0;
+            if (dim->len == 0) {
+              if ((ret = nc4_find_dim_len(dim_grp, dimid, &lenp)))
+                 return ret;
+              if (h5->no_write == NC_TRUE) {
+                dim->len = *lenp;
+              }
+            }
+            else {
+              *lenp = dim->len;
+            }
             if ((ret = nc4_find_dim_len(dim_grp, dimid, &lenp)))
```

Comment:
Shouldn't this second call to nc4_find_dim_len() be removed?

Reply:
Yes, that should have been deleted as part of the patch.  The version I based timings off of does not have that line.